### PR TITLE
Clayton

### PR DESCRIPTION
**Pull request by marsha88**
Updated routing in server.js. Still need the ajax request route to be changed to '/calculate' in order for posting to work. The server will try to fetch data from the TDPAlg.js, so if this part is erroring for any reason comment out the TDPAlg call and just use the example array data provided below it.

### CODE REVIEW

File: TDPAlg.js [hunk 1]
```diff
@@ -1,82 +1,144 @@
...
-		var deficitVolDay;
-		var inflowVolDay = //from db;
-
-		if(soilMoistureDepthDay < 0.5*maxSoilMoisture){
-			irrigationVolDay = irrigationDepth * irrigationArea;
-			//CHECK WITH FRANKENB TO MAKE SURE THIS IF STATEMENT SHOULD BE NESTED.
+"use strict";
+
+var db = require('./db');
+module.exports = function(_pondVolSmallest, _pondVolLargest, _pondVolIncrement, _pondDepth,_pondDepthInitial, _maxSoilMoisture, _irrigationArea, _irrigationDepth, _availableWaterCapacity, _locationId){		//TODO: add last argument.
```
Comment (drewatk) [inline TDPAlg.js]: You can just use `exports` here

File: TDPAlg.js [hunk 2]
```diff
@@ -1,82 +1,144 @@
...
+
+
+		var irrigationVolDay = 0;
+		var deficitVolDay;	
+		var bypassFlowVolDay;
+
+		/* Need to query database for evalVolDay or read from inputted CSV */
+		var evapVolDay;
+		
+		var pondPrecipVolDay = (percipDay * pondArea);
```
Comment (drewatk) [inline TDPAlg.js]: Misspelled `precipDay`?

File: TDPAlg.js [hunk 3]
```diff
@@ -1,82 +1,144 @@
...
+		var soilMoistureDepthDay = (soilMoistureDepthDayPrev + precipDepthDay - evapDepthDay);
+
+
+		if(soilMoistureDepthDay < (0.5*_availableWaterCapacity)){
+			irrigationVolDay = _irrigationDepth * _irrigationArea;
 			if(irrigationVolDay > waterVolDay){
```
Comment (drewatk) [inline TDPAlg.js]: You deleted your definition of `waterVolDay` but are still using it

File: TDPAlg.js [hunk 4]
```diff
@@ -1,82 +1,144 @@
...
+		if(pondWaterVolDay > pondVol){
+			bypassFlowVolDay = pondWaterVolDay - pondVol;
+			pondWaterVolDay = pondVol;
 		}
 		else{
-			bypassFlow=0
+			bypassFlowVolDay=0
 		}
 
 		pondWaterDepthDay = waterVolDay/pondArea;
```
Comment (drewatk) [inline TDPAlg.js]: Add `var`

File: TDPAlg.js [hunk 5]
```diff
@@ -1,82 +1,144 @@
...
+
+		(Write Date, InflowVolDay, EvaporationVolDay, SeepageVolDay, IrrigationVolDay, BypassVolDay, PondWaterDepthDay)
+
+		***************************************************************************************************************
+		*/
 
-		//update the prev
+		//update the (day-1) variables
 		soilMoistureDepthDayPrev = soilMoistureDepthDay;
 		waterVolDayPrev = waterVolDay;
```
Comment (drewatk) [inline TDPAlg.js]: should this be `pondWaterVolDayPrev`?

File: TDPAlg.js [hunk 6]
```diff
@@ -1,82 +1,144 @@
...
+		*/
+
+
+		/*******Calculate annual values*******/
+
+		inflowVolYear += inflowVolDay;	
+		evapVolYear += evapVolDay; 		
+		seepageVolYear += seepageVolDay;	
+		irrigationVolYear += irrigationVolDay;	
+		bypassVolYear += bypassVolDay
```
Comment (drewatk) [inline TDPAlg.js]: never defined `bypassVolDay`

File: TDPAlg.js [hunk 7]
```diff
@@ -1,82 +1,144 @@
...
+
+
+		var irrigationVolDay = 0;
+		var deficitVolDay;	
+		var bypassFlowVolDay;
+
+		/* Need to query database for evalVolDay or read from inputted CSV */
+		var evapVolDay = 1;	//temporary value for testing otuput.
+		
+		var pondPrecipVolDay = (percipDepthDay * pondArea);
```
Comment (drewatk) [inline TDPAlg.js]: never defined `percipDepthDay`
Also misspelled

File: TDPAlg.js (hunk 6 above)
Comment (drewatk) [inline TDPAlg.js]: Also semicolon

File: TDPAlg.js [hunk 8]
```diff
@@ -1,82 +1,144 @@
...
-		if(WaterVolDay > PondVol){
-			bypassFlowVolDay= waterVolDay – pondVol;
-			waterVolDay = pondVol;
+		if(pondWaterVolDay > pondVol){
+			bypassFlowVolDay = pondWaterVolDay - pondVol;
+			pondWaterVolDay = pondVol;
 		}
 		else{
-			bypassFlow=0
+			bypassFlowVolDay=0
```
Comment (drewatk) [inline TDPAlg.js]: semicolon

File: TDPAlg.js (hunk 4 above)
Comment (marsha88) [inline TDPAlg.js]: That was actually declared above. No idea why it picked this up aside from the no semi colon.

**Pull request merged**